tools/gitignore/gitignore.py

```diff
@@ -269,7 +269,7 @@ def filter(self,
                             target.append(item)
 
             dirnames[:] = keep_dirs
-            assert ".git" not in dirnames
+            assert not any(".git" == name for name, _ in dirnames)
```

Comment:
This assert has been guaranteed to never fail since 4011c5f3456 due to asserting `str in List[Tuple[str, T]]`, which always returns False.